src/ApiCallable.php

```diff
@@ -111,12 +112,34 @@ private static function setPageStreaming($callable, $pageStreamingDescriptor)
         return $inner;
     }
 
+    private static function setCustomHeader($callable, $headerDescriptor)
+    {
+        $inner = function() use ($callable, $headerDescriptor) {
+            $params = func_get_args();
+            if (count($params) != self::GRPC_CALLABLE_PARAM_COUNT ||
+                gettype($params[self::GRPC_CALLABLE_METADATA_INDEX]) != 'array') {
+                throw new InvalidArgumentException('Metadata argument is not found.');
+            } else {
+                $metadata = $params[self::GRPC_CALLABLE_METADATA_INDEX];
+                $params[self::GRPC_CALLABLE_METADATA_INDEX] =
+                    array_merge($metadata, $headerDescriptor->getHeader());
+                call_user_func_array($callable, $params);
+            }
+        };
+        return $inner;
+    }
+
     /**
      * @param \Grpc\BaseStub $stub the gRPC stub to make calls through.
      * @param string $methodName the method name on the stub to call.
      * @param \Google\GAX\CallSettings the call settings to use for this call.
+     * @param \Google\GAX\PageStreamingDescriptor
+     *     the descriptor used for page-streaming. Default is null.
+     * @param \Google\GAX\AgentHeaderDescriptor
+     *     the descriptor used for creating GAPIC header. Default is null.
      */
-    public static function createApiCall($stub, $methodName, CallSettings $settings)
+    public static function createApiCall($stub, $methodName, CallSettings $settings,
+        $pageStreamingDescriptor = null, $headerDescriptor = null)
```

Comment:
I think I would prefer to take the descriptors as an options array, since the parameter list is getting a little long.